mono/mini/memory-access.c

```diff
@@ -65,7 +65,8 @@ mini_emit_memset (MonoCompile *cfg, int destreg, int offset, int size, int val,
 
 	//Unaligned offsets don't naturaly happen in the runtime, so it's ok to be conservative in how we copy
 	//We assume that input src and dest are be aligned to `align` so offset just worsen it
-	int offsets_mask = offset & 0x7; //we only care about the misalignment part
+	int offsets_mask;
+	offsets_mask = offset & 0x7; //we only care about the misalignment part
```

Comment:
is that really needed?

Reply:
I really don't understand these changes, they generate the exact same machine code. Can this c++ feature be turned off somehow ?

Reply:
No. The language is the language. There is some choice of compiler & C++98 vs. C++11 etc. but I don’t believe this part is negotiable. It makes some sense — if you goto around initialization, what is the value of the variable? Well, it’d be nice if the compiler noticed it isn’t used, but that’d require always doing an optimizer’s analysis.
The C code seems a little dubious — looks always initialized but is not.

Reply:
So actually Visual C++ does allow this presently by default, maybe later under -fpermissive. But I'm using clang lately and I doubt there is a switch. Usually clang is very free to tell me switches to change its behavior but there wasn't one shown here.